datalad_redcap/export_form.py

```diff
@@ -100,22 +101,41 @@ class ExportForm(Interface):
         save=nosave_opt,
     )
 
+    _validator_ = EnsureCommandParameterization(
+        dict(
+            url=EnsureURL(required=["scheme", "netloc", "path"]),
+            forms=EnsureListOf(str),
+            outfile=EnsurePath(),
+            dataset=EnsureDataset(installed=True, purpose="export REDCap form"),
+            survey_fields=EnsureBool(),
+            credential=EnsureStr(),
+            message=EnsureStr(),
+            save=EnsureBool(),
+        ),
+    )
+
     @staticmethod
     @datasetmethod(name="export_redcap_form")
     @eval_results
     def __call__(
         url: str,
         forms: List[str],
-        outfile: str,
-        dataset: Optional[Union[Dataset, str]] = None,
+        outfile: Path,
+        dataset: Optional[DatasetParameter] = None,
```

Comment:
I have settled on the following pattern in my code:

```suggestion
        dataset: Dataset | None = None,
```

it requires

```py
from __future__ import annotations
```

at the top of the file.

I do not think we have standards on annotation style, not do I claim to be an authority on the topic. Just thought there is value in mentioning it.

Reply:
Thanks! I am using this project to learn more about the things I didn't have much practice with, so there's definitely value. I internalized the logic of `Optional`, but I also see that `X | Y` is newer and was proposed as an alternative (I even ran into a [no-optional](https://github.com/Kludex/no-optional) linter).

I will leave `Optional` for now, but will probably also switch later.